pennylane/ops/qubit/matrix_ops.py

```diff
@@ -332,3 +334,134 @@ def _controlled(self, control):
 
     def label(self, decimals=None, base_label=None, cache=None):
         return super().label(decimals=decimals, base_label=base_label or "U", cache=cache)
+
+
+class BlockEncode(Operation):
```

Comment:
You should be able to add an `adjoint` method which returns the adjoint of the block encoding by simply instantiating the class with A^dagger instead. I am 90% sure that should work, but also worth double checking.

Reply:
Not blocking, but a nice to have when building new operators to make them compatible